### ISSUE TITLE
E3SM config files do not take driver into account

### ISSUE THREAD

**Issue by jgfouca** [ESMCI/cime#2861]
Now that multiple drivers/couples are available, E3SM config files need to take the driver of the current case into account instead of assuming mct. The file that needs to be modified is:

```
config/e3sm/config_files.xml
```

As an example, for CESM, the value of CONFIG_CPL_FILE is:

```
$COMP_ROOT_DIR_CPL/cime_config/config_component.xml
```

and for E3SM the value is:

```
$CIMEROOT/src/drivers/mct/cime_config/config_component.xml
```

Looking at the two values, you can see that E3SM is hardcoded to mct while CESM is flexible. There are many entries that will need to be made flexible like this in E3SM.

Once this is done, the test:

```
 TestCreateNewcase.test_m_createnewcase_alternate_drivers
```

... should PASS.

**Comment by jgfouca** [comment]
@rljacob , I assigned to you so you can assign to one of our newer CIME developers, this is a good first issue.

This is fairly high priority since it's causing test failures. If no one will be able to get to it soon, then assign to me.

**Comment by sarich** [comment]
@jgfouca I think I have the config file fixed in branch sarich/update-e3sm-config-driver, it passes for moab anyway.
It's not passing for --driver nuopc. It needs a file src/drivers/nuopc/cime_config/config_component_e3sm.xml

I tried copying the file from mct, but I'm still getting errors and I'm not sure where to go from here:
```
Traceback (most recent call last):
  File "./case.setup", line 62, in <module>
    _main_func(__doc__)
  File "./case.setup", line 59, in _main_func
    case.case_setup(clean=clean, test_mode=test_mode, reset=reset)
  File "/sandbox/sarich/cime/scripts/Tools/../../scripts/lib/CIME/case/case_setup.py", line 247, in case_setup
    run_and_log_case_status(functor, phase, caseroot=caseroot)
  File "/sandbox/sarich/cime/scripts/Tools/../../scripts/lib/CIME/utils.py", line 1680, in run_and_log_case_status
    rv = func()
  File "/sandbox/sarich/cime/scripts/Tools/../../scripts/lib/CIME/case/case_setup.py", line 231, in <lambda>
    functor = lambda: _case_setup_impl(self, caseroot, clean, test_mode, reset)
  File "/sandbox/sarich/cime/scripts/Tools/../../scripts/lib/CIME/case/case_setup.py", line 215, in _case_setup_impl
    case.create_namelists()
  File "/sandbox/sarich/cime/scripts/Tools/../../scripts/lib/CIME/case/preview_namelists.py", line 89, in create_namelists
    run_sub_or_cmd(cmd, (caseroot), "buildnml", (self, caseroot, compname), case=self)
  File "/sandbox/sarich/cime/scripts/Tools/../../scripts/lib/CIME/utils.py", line 360, in run_sub_or_cmd
    getattr(mod, subname)(*subargs)
  File "/sandbox/sarich/cime/src/drivers/nuopc/cime_config/buildnml", line 407, in buildnml
    _create_drv_namelists(case, infile, confdir, nmlgen, files)
  File "/sandbox/sarich/cime/src/drivers/nuopc/cime_config/buildnml", line 77, in _create_drv_namelists
    start_type = nmlgen.get_value('start_type')
  File "/sandbox/sarich/cime/scripts/Tools/../../scripts/lib/CIME/nmlgen.py", line 200, in get_value
    return self._to_python_value(name, self._namelist.get_value(name))
  File "/sandbox/sarich/cime/scripts/Tools/../../scripts/lib/CIME/nmlgen.py", line 146, in _to_python_value
    var_type, _, var_size, = self._definition.split_type_string(name)
  File "/sandbox/sarich/cime/scripts/Tools/../../scripts/lib/CIME/XML/namelist_definition.py", line 207, in split_type_string
    type_string = self._entry_types[name]
KeyError: 'start_type'
```

**Comment by jgfouca** [comment]
@jedwards4b , any thoughts on this?

**Comment by jedwards4b** [comment]
start_type is in the namelist_definition file not the config_component file - I'll need to look at this a little more.

**Comment by jgfouca** [comment]
@jedwards4b , @sarich any updates? This is the last unresolved nuopc issue for e3sm (a fairly minor issue since e3sm is not using anything but mct).

**Comment by jedwards4b** [comment]
I dropped the ball, looking into it now.

**Comment by jedwards4b** [comment]
Okay I can reproduce the problem, that's a start.

**Comment by jgfouca** [comment]
Once this goes in, I will retry the CIME update to E3SM.

**Comment by sarich** [comment]
I don't have any updates. I think this branch can be put into a PR, with another issue made to  fix the E3SM/nuopc errors.

**Comment by jedwards4b** [comment]
@sarich I have an additional fix - can I just push to this branch?

**Comment by sarich** [comment]
Yes, please do
